example_queries/get_service_name.rs.ref

```diff
@@ -2,50 +2,74 @@ use rpc_lib::rpc::Rpc;
 use indexmap::map::IndexMap;
 use petgraph::graph::NodeIndex;
 use petgraph::Graph;
-use petgraph::Outgoing;
+use petgraph::Incoming;
 use graph_utils::utils;
 use graph_utils::iso::find_mapping_shamir_centralized;
+use serde::{Serialize, Deserialize};
+extern crate serde_json;
 
 pub type CodeletType = fn(&Filter, &Rpc) -> Option<Rpc>;
 
-// user defined functions:
 
+#[derive(Debug, Serialize, Deserialize)]
```

Comment:
At this point this structures can be part of a lib? Or do they always have to be newly generated?

Reply:
They could be a part of a lib.  But given that it's about 40 lines, and makes for easy access when you're referencing them, I don't see the point.  Plus, we're going to have to change the contents once we do distributed isomorphism in order to propagate partial isomorphism information, and then we'd be keeping multiple library versions around which I think might get confusing.  Is there any particular reason you'd like it to be a library, besides cleanliness?

Reply:
Mostly easy of use, maintainability, and for testing. The filter should have as little static code as possible (basically only items that are auto generated) because the more we throw in there the harder it is to debug and quickly understand those programs. I can take care of this once everything has stabilized and we are actually deploying these filters.